packages/policy/src/changer.ts

```diff
@@ -55,108 +51,95 @@ async function cachedGet(url: string) {
   return contents;
 }
 
-/**
- * Submit a pull request to the target repository to add a CODE_OF_CONDUCT.md.
- * @param owner Repostiory owner - ex: "googleapis"
- * @param repo Repository name - ex: "repo-automation-bots"
- * @param octokit Pre-authenticated octokit instance
- */
-export async function addCodeOfConduct(repo: GitHubRepo, octokit: Octokit) {
-  // first, make sure there's no open PR for this
-  const title = 'chore: add a Code of Conduct';
-  const prs = await octokit.search.issuesAndPullRequests({
-    q: `repo:${repo.full_name} "${title}" in:title is:open`,
-  });
-  if (prs.data.total_count > 0) {
-    return;
+export class Changer {
+  private octokit: Octokit;
+  private repo: GitHubRepo;
+  private owner: string;
+  private name: string;
+  private prsCache?: PullRequests;
+
+  constructor(octokit: Octokit, repo: GitHubRepo) {
+    this.octokit = octokit;
+    this.repo = repo;
+    const [owner, name] = repo.full_name.split('/');
+    this.owner = owner;
+    this.name = name;
   }
 
-  // fetch the CoC from `googleapis/.github`
-  const content = await getCoC();
-
-  // submit the PR
-  const changes: Changes = new Map([
-    [
-      'CODE_OF_CONDUCT.md',
-      {
-        content,
-        mode: '100644',
-      },
-    ],
-  ]);
-
-  const [owner, name] = repo.full_name.split('/');
-  await createPullRequest(octokit, changes, {
-    title,
-    message: title,
-    description: 'add a code of conduct',
-    upstreamOwner: owner,
-    upstreamRepo: name,
-    fork: false,
-    primary: repo.default_branch,
-    retry: 0,
-    branch: `policy-bot-${uuid()}`,
-  });
-}
-
-/**
- * Submit a pull request to the target repository to add a SECURITY.md.
- * @param repo Repository name - ex: "repo-automation-bots"
- * @param octokit Pre-authenticated octokit instance
- */
-export async function addSecurityPolicy(repo: GitHubRepo, octokit: Octokit) {
-  // first, make sure there's no open PR for this
-  const title = 'chore: add SECURITY.md';
-  const prs = await octokit.search.issuesAndPullRequests({
-    q: `repo:${repo.full_name} "${title}" in:title is:open`,
-  });
-  if (prs.data.total_count > 0) {
-    return;
+  /**
+   * Check to see if the current repository has an open pull request with the given title
+   * @param title The title to search for
+   * @returns
+   */
+  private async hasMatchingPR(title: string) {
+    if (!this.prsCache) {
+      this.prsCache = await this.octokit.paginate(this.octokit.pulls.list, {
+        owner: this.owner,
+        repo: this.name,
+        state: 'open',
+        per_page: 100,
+      });
+    }
+    const matches = this.prsCache.filter(x => x.title === title);
```

Comment:
Would `prsCache` have the potential to cause the process to run out of memory for repos that have a lot of open PRs? This may not be a problem atm - I'll share this snippet in case it becomes one:

```ts
const params = {
  owner: this.owner,
  repo: this.name,
  state: 'open',
  per_page: 100,
};

for await (const response of octokit.paginate.iterator(this.octokit.pulls.list, params)) {
  if (response.title === title) {
    return true;
  }
}

return false;
```

Reply:
This is handy, and usually I'd totally agree with you.  In this particular case though, I'm caching the total set of open PRs on the repository because I want to go back through them multiple times to check for other open PRs.  The caching happens since we're already submitting 2 different types of PR in here, each where I have to check for the title.  If a repository has more than a few dozen open pull requests, they're probably having a bad day anyhow :)